aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -98,124 +108,275 @@ module AptosFramework::Stake {
+        account: &signer,
+        pool_address: address,
+        consensus_pubkey: vector<u8>,
+    ) acquires StakePool, ValidatorConfig {
+        let stake_pool = borrow_global<StakePool>(pool_address);
+        assert!(Signer::address_of(account) == stake_pool.operator_address, ENOT_OPERATOR);
+
+        assert!(exists<ValidatorConfig>(pool_address), Errors::not_published(EVALIDATOR_CONFIG));
+        let validator_info = borrow_global_mut<ValidatorConfig>(pool_address);
         validator_info.consensus_pubkey = consensus_pubkey;
     }
 
-    /// Initialize validator set to the core resource account.
-    public fun initialize_validator_set(account: &signer, minimum_stake: u64, maximum_stake: u64) {
-        SystemAddresses::assert_core_resource(account);
-        move_to(account, ValidatorSet {
-            consensus_scheme: 0,
-            minimum_stake,
-            maximum_stake,
-            active_validators: Vector::empty(),
-            pending_active: Vector::empty(),
-            pending_inactive: Vector::empty(),
-        });
+    /// Similar to unlock_with_cap but will use ownership capability from the signing account.
+    public(script) fun increase_lockup(
+        account: &signer,
+        new_locked_until_secs: u64,
+    ) acquires OwnerCapability, StakePool {
+        let account_addr = Signer::address_of(account);
+        let ownership_cap = borrow_global<OwnerCapability>(account_addr);
+        increase_lockup_with_cap(account_addr, ownership_cap, new_locked_until_secs);
     }
 
-    /// Initiate by the validator info owner
-    public(friend) fun join_validator_set(account: &signer) acquires StakePool, ValidatorConfig, ValidatorSet {
-        let addr = Signer::address_of(account);
-        let stake_pool = borrow_global<StakePool>(addr);
+    /// Unlock from active delegation, it's moved to pending_inactive if locked_until_secs < current_time or
+    /// directly inactive if it's not from an active validator.
+    public fun increase_lockup_with_cap(
+        pool_address: address,
+        owner_cap: &OwnerCapability,
+        new_locked_until_secs: u64,
+    ) acquires StakePool {
+        assert!(owner_cap.pool_address == pool_address, ENOT_OWNER);
+        validate_lockup_time(new_locked_until_secs);
+
+        let stake_pool = borrow_global_mut<StakePool>(pool_address);
+        assert!(new_locked_until_secs > stake_pool.locked_until_secs, ELOCK_TIME_TOO_SHORT);
+        stake_pool.locked_until_secs = new_locked_until_secs;
+    }
+
+    /// Request to join the validator set. Can only be called after calling `register_validator_candidate`.
+    /// If the validator has the required stake (more than minimum and less than maximum allowed), they will be
+    /// added to the pending_active queue. All validators in this queue will be added to the active set when the next
+    /// epoch starts (eligibility will be rechecked).
+    public(friend) fun join_validator_set(
```

Comment:
I feel this should be on operators instead of owners, like only join after their node is running properly.

Reply:
Makes sense. Updated.